drivers/display/ssd1306.c

```diff
@@ -48,22 +49,56 @@ LOG_MODULE_REGISTER(ssd1306, CONFIG_DISPLAY_LOG_LEVEL);
 
 struct ssd1306_data {
 	const struct device *reset;
-	const struct device *i2c;
+	const struct device *bus;
+#if DT_INST_ON_BUS(0, spi)
+	struct spi_cs_control cs_ctrl;
+	struct spi_config spi_config;
+	const struct device *data_cmd;
+#endif
 	uint8_t contrast;
 	uint8_t scan_mode;
 };
 
-static inline int ssd1306_write_i2c(const struct device *dev,
+#if DT_INST_ON_BUS(0, i2c)
+static inline int ssd1306_write_bus(const struct device *dev,
 				    uint8_t *buf, size_t len, bool command)
 {
 	struct ssd1306_data *driver = dev->data;
 
-	return i2c_burst_write(driver->i2c, DT_INST_REG_ADDR(0),
+	return i2c_burst_write(driver->bus, DT_INST_REG_ADDR(0),
 			       command ? SSD1306_CONTROL_ALL_BYTES_CMD :
 					 SSD1306_CONTROL_ALL_BYTES_DATA,
 			       buf, len);
 }
 
+#elif DT_INST_ON_BUS(0, spi)
+
+static inline int ssd1306_write_bus(const struct device *dev,
+				    uint8_t *buf, size_t len, bool command)
+{
+	struct ssd1306_data *driver = dev->data;
+	int errno;
+
+	gpio_pin_set(driver->data_cmd, DT_INST_GPIO_PIN(0, data_cmd_gpios),
+		     command ? 0 : 1);
+	struct spi_buf tx_buf = {
+		.buf = buf,
+		.len = len
+	};
+
+	struct spi_buf_set tx_bufs = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	errno = spi_write(driver->bus, &driver->spi_config, &tx_bufs);
+	gpio_pin_set(driver->data_cmd,
```

Comment:
Is it necessary to reset DC gpio?

Reply:
No, since the pin usually is not pulled in any direction, it is perfectly safe to let it be.
I removed the command.